### ISSUE TITLE
Training loss is 0

### ISSUE THREAD

**Issue by atlefk14** [AlexeyAB/darknet#3773]
During training I get an Training loss of 0.0 (after 600 epochs), which in my experience suggests that the network is overfitting. I am training the network as suggested in https://github.com/AlexeyAB/darknet#how-to-train-to-detect-your-custom-objects. 
What I find weird is that my map score is always 0 during the training. The split between the classes are equal in the training and validation set.  

My custom dataset has four different classes. 

![chart](https://user-images.githubusercontent.com/32202809/63209863-a1bc6400-c0e6-11e9-94a7-16bb8e9b9d85.png)

I added my training loss graph for illustration purposes. 

Do you have any suggestions?

**Comment by AlexeyAB** [comment]
Run training command with flag ` -show_imgs`

**Comment by atlefk** [comment]
You want me to keep training even tho the loss has reached 0? 

The bounding box are not supposed to show up, when I use -show_imgs? I get the pictures just fine, but cannot see a bounding box.

**Comment by AlexeyAB** [comment]
Your labels are wrong

**Comment by atlefk14** [comment]
I assume that you suspect that by previous experience. I have check the class labels, and they are indeed correct. Assuming that the classes starts at 0 (as written in the documentation). 1,0.025,0.431640625,0.14791666666666667,0.1203125 is a example of how a object of class 2 is defined.  Where the first is the class, then x centre, y centre, width, height.

**Comment by atlefk14** [comment]
I think you are right tho, the scripts the converts from another format to this one, was using the shape of the images incorrectly. I will generate new files and try again. Thanks!

**Comment by atlefk14** [comment]
![chart](https://user-images.githubusercontent.com/32202809/63259082-bd0da780-c27d-11e9-9d21-b50e4c81d1e4.png)

This is how it looks like now. map is still 0, the loss is 0. I have trained other networks with the data and labels, the only thing which is different is that they used min/max values, and this uses centre values.

**Comment by AlexeyAB** [comment]
Your labels are wrong

**Comment by atlefk14** [comment]
Okei, can you elaborate a little bit on which part of the labels you suspect are wrong? 
In my original dataset, classes range from 1-4, they have been mapped from 0-3 instead, as this implementation requires that. 

My labels file contains these.

Ccu_Basket
Ccu_ClosedTop
Ccu_GasTank 
Ccu_Skip

Here is an example of a GasTank (I have checked the actual image)
2,0.81953125,0.78125,0.215625,0.24166666666666667

The image is 640 x 480. 
Meaning the x-centre is ish = 518, and y-centre is 375

When I draw a red dot in the centre using the above mentioned coordinates I get this; which suggest that these coordinates are correct. 
![test](https://user-images.githubusercontent.com/32202809/63304606-e7924b80-c2e3-11e9-8603-e229aeeed4b8.png)

The width is: 134.4 and the height is 115.2 (I assume that it should be the entire width and height, and not just from the centre) 

Btw, the dataset is kinda unbalanced, maybe that has an impact? There are 25k images in total, 80% is used for training and 20% for validation, the split between the classes in the subsets are equal. 

I do not know where the error is, any suggestions would be appreciated. 
Thanks for the assist!

**Comment by AlexeyAB** [comment]
> The bounding box is not supposed to show up when I use **-show_imgs? I get the pictures just fine, but cannot see a bounding box.**

**Comment by atlefk14** [comment]
Does that mean that the bounding boxes are supposed to show up when the -show_imgs flag is on?

**Comment by AlexeyAB** [comment]
Yes
